### ISSUE TITLE
ssh `recvall` is non-interruptible

### ISSUE THREAD

**Issue by zachriggle** [Gallopsled/pwntools#459]
This is interruptible with a Ctrl+C

```
from pwn import *
r = remote('google.com',80)
r.recvall()
```

This is not

```
from pwn import *
s = ssh('demouser','localhost').run('bash')
s.recvall()
```

This issue is also present in the `2.2.0` tag.

**Comment by zachriggle** [comment]
It looks like this may be a symptom of `ssh.wait`.

```
from pwn import *
s = ssh('demouser','localhost').run('bash')
s.wait()
EOF
[*] Connecting to localhost on port 22: 
[+] Connecting to localhost on port 22: Done
[*] Opening new channel: 'bash': 
[+] Opening new channel: 'bash': Done
^C^C^C^C^C^C^C
```

**Comment by zachriggle** [comment]
Even further, this looks like a symptom of the fact that this is uninterruptible:

```
import threading
threading.Event().wait()
```

**Comment by zachriggle** [comment]
Looks like we'd have to change the behavior of `ssh_channel.recv_raw` as well as that of `ssh_channel.wait` in order to circumvent this issue.

`paramiko.buffered_pipe.BufferedPipe.read` will block using an `Event` in the same manner, e.g. when calling `recvall`.

**Comment by zachriggle** [comment]
This should be fixed now that #460 is merged
